### ISSUE TITLE
callable-types should handle `new(): T` just like `(): T`

### ISSUE THREAD

**Issue by andy-ms** [palantir/tslint#4291]
### Bug Report

- __TSLint version__: 5.11.0
- __TypeScript version__: 3.2.0-dev.20181114
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
interface I0 { (): void; }
interface I1 { new (): void; }
```

with `tslint.json` configuration:

```json
{ "extends": "tslint:latest" }
```

#### Actual behavior

Lint failure at `I0` but not at `I1`.

#### Expected behavior

Lint failure at both, since this can be changed to:
```ts
type I0 = () => void;
type I1 = new () => void;
```

**Comment by JoshuaKGoldberg** [comment]
Agreed, thanks for reporting this @andy-ms! I'd be happy to express merge a PR for this.

**Comment by JoshuaKGoldberg** [comment]
Fixed by #4352! ✨
